crates/bevy_ecs/src/system/mod.rs

```diff
@@ -435,4 +437,31 @@ mod tests {
         let d_id = world.components().get_id(TypeId::of::<D>()).unwrap();
         assert_eq!(conflicts, vec![b_id, d_id]);
     }
+
+    #[test]
+    #[allow(clippy::too_many_arguments)]
+    fn can_have_16_parameters() {
+        fn sys_x(
```

Comment:
We also want to check big tuples used as system parameters, such as 
```rust
#[derive(Default)]
struct A<const X: usize>();
type P<const X: usize> = Local<A<X>>;
fn sys_y(_: (Local<T0>, Local<T1>, ..., Local<T16>), Query<&T> )
```

Reply:
while trying to add this test, I found another issue that's also present on main, `Local` can only be the first parameter of a system or the first element of a tuple. checking how to fix this